### ISSUE TITLE
Undefined class constant 'OPT_SCAN' 

### ISSUE THREAD

**Issue by richwandell** [phpredis/phpredis#466]
The following code taken strait from the readme file produces an undefined class constant error.

```php
$redis = new Redis();

$redis->connect('127.0.0.1', '22124');

$it = NULL; /* Initialize our iterator to NULL */
$redis->setOption(Redis::OPT_SCAN, Redis::SCAN_RETRY); /* retry when we get no keys back */
while($arr_keys = $redis->scan($it)) {
    foreach($arr_keys as $str_key) {
        echo "Here is a key: $str_key\n";
    }
    echo "No more keys to scan!\n";
}
```

**Comment by dinamic** [comment]
Most probably you are running some old version. You can use the following command to supply us the version of the redis extension you have:

```php -i | grep Redis
Redis Support => enabled
Redis Version => 2.2.4```

Here's the output from mine, which does not support ```OPT_SCAN```, because the version is way too old.

```
$ php -i | grep Redis
Redis Support => enabled
Redis Version => 2.2.4
```

**Comment by dustinalan** [comment]
I ran into the same problem, updated the version and no problemo. 

Ticket can be closed IMO.

    $ php -i | grep Redis
    Redis Support => enabled
    Redis Version => 2.2.4

**Comment by dustinalan** [comment]
For those on Ubuntu, I added some instructions on how to upgrade to version 2.2.5+ 
http://stackoverflow.com/a/30490869/2085469

**Comment by izolate** [comment]
@michael-grunder Is there any way to update the `php5-redis` package on Apt? Latest package available is currently an old 2.2.4
